### PR TITLE
Clarify why TextVectorization works on CPU

### PR DESCRIPTION
**Pull request by 8bitmp3**
As discussed with @fchollet @mattdangerw @MarkDaoust 

This PR introduces the following improvements:

- Clarify that the Keras `TextVectorization` layer should be used as part of a `tf.data` input pipeline (runs only on CPUs, won't utilize hardware acceleration if included inside a `Model`, affecting performance).
- Perform some minor linting.

---

- Guide: Working with preprocessing layers (https://keras.io/guides/preprocessing_layers/, https://www.tensorflow.org/guide/keras/preprocessing_layers)
- Section: Preprocessing data before the model or inside the model

### CODE REVIEW

**Review by fchollet — COMMENTED** [review]
LGTM, thanks for the PR. Please regenerate the `ipynb` and `md` files.

**Review by mattdangerw — APPROVED** [review]
Thanks!

File: guides/md/preprocessing_layers.md [hunk 1]
```diff
@@ -614,6 +638,7 @@ print("Model output:", test_output)
 
 <div class="k-default-codeblock">
 ```
+WARNING:tensorflow:5 out of the last 1567 calls to <function PreprocessingLayer.make_adapt_function.<locals>.adapt_step at 0x7f80ec464a60> triggered tf.function retracing. Tracing is expensive and the excessive number of tracings could be due to (1) creating @tf.function repeatedly in a loop, (2) passing tensors with different shapes, (3) passing Python objects instead of tensors. For (1), please define your @tf.function outside of the loop. For (2), @tf.function has reduce_retracing=True option that can avoid unnecessary retracing. For (3), please refer to https://www.tensorflow.org/guide/function#controlling_retracing and https://www.tensorflow.org/api_docs/python/tf/function for  more details.
```
Comment (8bitmp3) [inline guides/md/preprocessing_layers.md:641]: @mattdangerw @fchollet  Note the warnings in the output after regenerating `.md`

File: guides/md/preprocessing_layers.md [hunk 2]
```diff
@@ -116,6 +116,14 @@ print("Features std: %.2f" % (normalized_data.numpy().std()))
 
 <div class="k-default-codeblock">
 ```
+2022-06-15 15:02:07.223345: W tensorflow/stream_executor/platform/default/dso_loader.cc:64] Could not load dynamic library 'libcudart.so.11.0'; dlerror: libcudart.so.11.0: cannot open shared object file: No such file or directory
```
Comment (8bitmp3) [inline guides/md/preprocessing_layers.md:119]: @mattdangerw @fchollet Note tha new messages in the output after regerating the Markdown file

**Review by fchollet — APPROVED** [review]
We'll remove the warnings in post. Thanks!
